### DISCUSSION TITLE
edit scaledobject

### DISCUSSION THREAD

**Discussion by gal-aflalo** [kedacore/keda#4660] [Q&A / Need Help] [answered]
### Report

I created a scaledobject to cpu scale, and when trying to edit it with 'kubectl edit scaledobjects <sacledobject_name>', its not working and I'm getting an error.

### Expected Behavior

minReplicaCount is changed to 1 and update the hpa.

### Actual Behavior

minReplicaCount is not changed.
error: scaledobjects.keda.sh "<sacledobject_name>" could not be patched: Timeout: request did not complete within requested timeout - context deadline exceeded
You can run `kubectl replace -f /tmp/kubectl-edit-3879380713.yaml` to try this update again.

### Steps to Reproduce the Problem

1. apply scaledobject with cpu scaler. 
2. kubectl edit scaledobject minReplicaCount.

### Logs from KEDA operator

```
example
```

### KEDA Version

2.10.1

### Kubernetes Version

1.25

### Platform

Amazon Web Services

### Scaler Details

cpu

**Comment by tomkerkhove** [comment]
This sounds more like a general Kubernetes issue where it is timing out talking to the control-plane, no?

**Reply by gal-aflalo** [reply]
why does it happen? 
Is it possible to edit existing scaledobject?

**Reply by zroubalik** [reply]
yes, it is a standart k8s resource

**Reply by gal-aflalo** [reply]
figured it out, it seems there is a slowness issue on rke, and also editing the scaledobjects is not working as described. 
When tested it on eks it worked fine and no slowness was felt. 
Is it a known issue? do you have an idea?

**Reply by gal-aflalo** [reply]
another update, checked on another eks and it also does not work, so I'm not sure what are the differences and why.

**Reply by MuriukiPM** [reply]
Adding a security group rule allowing tcp ingress for port 9443 from control plane to the node security group fixed this for me.

**Reply by jagtapa** [reply]
I believe this information is extremely important, especially since we encountered a similar issue ourselves and couldn't locate any guidance in the documentation. After spending a significant amount of time debugging, we managed to identify the issue and whitelist it, ultimately resolving our problem. 

@zroubalik: Could you please verify if this should be included in Keda's firewall section?

**Reply by zroubalik** [reply]
@jagtapa could you please open a docs PR and add information about this into the Troubleshooting section? I agree this is an important infromation. Thanks!
https://keda.sh/docs/2.13/troubleshooting/

**Reply by velann21** [reply]
@jagtapa What is the exact fix for this issue? I am facing the same issue, timing out while editing the existing scaledobject

**Reply by velann21** [reply, marked as answer]
I was able to solve this problem by allow ingress network policy on keda installed namespace to allow port 9443.

**Reply by jagtapa** [reply]
@velann21 we were able to fix this issue by allowing tcp ingress for port 9443 from Control Plane to the GKE Node Pools
